src/udev/udev-builtin-net_id.c

```diff
@@ -835,14 +835,22 @@ static int names_devicetree(sd_device *dev, const char *prefix, EventMode mode)
         if (!streq(prefix, "en"))
                 return -EOPNOTSUPP;
 
-        /* check if our direct parent has an of_node */
-        r = sd_device_get_parent(dev, &parent);
-        if (r < 0)
-                return log_device_debug_errno(dev, r, "Failed to get parent device: %m");
 
-        r = sd_device_new_child(&ofnode_dev, parent, "of_node");
-        if (r < 0)
-                return log_device_debug_errno(parent, r, "Failed to get 'of_node' child device: %m");
+        /* check if the device itself has an of_node */
+        if (naming_scheme_has(NAMING_DEVICETREE_PORT_ALIASES))
+                r = sd_device_new_child(&ofnode_dev, dev, "of_node");
+        else
+                r = -1;
+        if (r < 0) {
+                /* check if our direct parent has an of_node as a fallback */
+                r = sd_device_get_parent(dev, &parent);
```

Comment:
Not a matter, but I"d like to add a debugging log for the first `_new_child()` call.
```c
if (naming_scheme_has(NAMING_DEVICETREE_PORT_ALIASES)) {
        r = sd_device_new_child(&ofnode_dev, dev, "of_node");
        if (r < 0)
                log_device_debug_errno(dev, r, "Failed to get 'of_node' child device, ignoring: %m");
}
if (!ofnode_dev) {
        sd_device *parent;

        r = sd_device_get_parent(dev, &parent);
        ...
}
```

Reply:
Changed as suggested. I have also reworded the debug messages a bit, as an of_node is not a device.